src/com/dotmarketing/viewtools/FileTool.java

```diff
@@ -28,16 +36,20 @@ public void init(Object initData) {
 	public File getNewFile(){
 		return new File();
 	}
-	
+
 	public IFileAsset getFile(String identifier, boolean live){
+		return getFile(identifier, live, -1);
+	}
+
+	public IFileAsset getFile(String identifier, boolean live, long languageId){
 		Identifier id;
 		String p = null;
 		try {
-			id = APILocator.getIdentifierAPI().find(identifier);
+			id = identifierAPI.find(identifier);
 		    if(live){
-			  p = LiveCache.getPathFromCache(id.getURI(), id.getHostId());
+			  p = LiveCache.getPathFromCache(id.getURI(), id.getHostId(), languageId != -1 ? languageId : null);
 		    }else{
-			  p = WorkingCache.getPathFromCache(id.getURI(), id.getHostId());
+			  p = WorkingCache.getPathFromCache(id.getURI(), id.getHostId(), languageId != -1 ? languageId : null);
```

Comment:
Will, I know they are deprecated but for this logic FileTool.java + ContentletServices.java they are used on several places so I decided that it was not the best moment to refactor this with the release so close. Because we don't have enough time to QA all the features related to ContentletServices. We could create an issue for next release.

Also about the LanguageWebApi.getLanguage(request), the FileTool viewtool has application scope, so I don't have the request info in there.